skpr/mail
# Feat: add timeout

## Overview
Adding a timeout to skprmail avoidingfor unnecessarily long wait times with no errors or closures
## Implementation
There was a context created and passed to both the ses and the default method. The ses method automatically handles the timeout via the context whereas we've implemented a goroutine to check for context timeout and close the application.
## How to Test
Setup a mail server using `smtp-go` (https://github.com/emersion/go-smtp/blob/master/cmd/smtp-debug-server/main.go) package and add sleep at `/server.go:211` This will ensure that the smtp server is available but will not respond to a connection dial leading our skprmail to be hung until the timeout is hit.
## What is the rollout plan?
Merge and rollout
## Does this change need a blog post?
Maybe a blog pose around contexts and timeouts
## Link to Tickets
* <not sure>
